lms/djangoapps/courseware/tests/test_entrance_exam.py

```diff
@@ -0,0 +1,276 @@
+"""
+Tests use cases related to LMS Entrance Exam behavior, such as gated content access (TOC)
+"""
+from django.test.client import RequestFactory
+from django.test.utils import override_settings
+
+from courseware.model_data import FieldDataCache
+from courseware.module_render import get_module, toc_for_course
+from courseware.tests.factories import UserFactory
+from milestones import api as milestones_api
+from milestones.models import MilestoneRelationshipType
+from xmodule.modulestore.django import modulestore
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase, TEST_DATA_MOCK_MODULESTORE
+from xmodule.modulestore.tests.factories import CourseFactory, ItemFactory
+from util.milestones_helpers import generate_milestone_namespace, NAMESPACE_CHOICES
+
+
+@override_settings(MODULESTORE=TEST_DATA_MOCK_MODULESTORE)
+class EntranceExamTestCases(ModuleStoreTestCase):
+    """
+    Check that content is properly gated.  Create a test course from scratch to mess with.
+    """
+
+    def setUp(self):
+        """
+        Test case scaffolding
+        """
+        super(EntranceExamTestCases, self).setUp()
+        self.course = CourseFactory.create(
+            metadata={
+                'entrance_exam_enabled': True,
+            }
+        )
+        chapter = ItemFactory.create(
+            parent=self.course,
+            display_name='Overview'
+        )
+        ItemFactory.create(
+            parent=chapter,
+            display_name='Welcome'
+        )
+        ItemFactory.create(
+            parent=self.course,
+            category='chapter',
+            display_name="Week 1"
+        )
+        ItemFactory.create(
+            parent=chapter,
+            category='sequential',
+            display_name="Lesson 1"
+        )
+        ItemFactory.create(
+            category="instructor",
+            parent=self.course,
+            data="Instructor Tab",
+            display_name="Instructor"
+        )
+        self.entrance_exam = ItemFactory.create(
+            parent=self.course,
+            category="chapter",
+            display_name="Entrance Exam Section - Chapter 1"
+        )
+        self.exam_1 = ItemFactory.create(
+            parent=self.entrance_exam,
+            category='sequential',
+            display_name="Exam Sequential - Subsection 1",
+            graded=True,
+            metadata={'in_entrance_exam': True}
+        )
```

Comment:
I think if we want to test that the "quick exit" regarding when to recalculate score averages is working with the new inherited metadata, we'd also like a negative case: a sequential which does not have 'in_entrance_exam' set to false and has a problem. Then the test assertion would to make sure that submitting an answer on the problem that is not in an entrance exam does not call the score calculation function.